apis/v1alpha1/types.go

```diff
@@ -90,3 +94,30 @@ type ProviderConfigUsageList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []ProviderConfigUsage `json:"items"`
 }
+
+// ProviderConfig type metadata.
```

Comment:
Do you think having these parts below in `register.go` still might make them more visible?

Reply:
Not sure. The initial reason to have them in `types.go` was to be able to generate the file more easily. I think it makes it easier to see them as well, i.e. you can find everything related to that specific CRD under that file and you'd know this type metadata is always at the bottom of it.

Reply:
I see, I feel like if there are many types in `types.go` then it can become a little unwieldy to have to scroll all the way to the bottom. If it makes a meaningful difference in how easy it is to generate then it is probably worth it though.

Reply:
The main thing in the code generator is that files that aggregate information from multiple sources are harder to build. For example, [here](https://github.com/crossplane-contrib/terrajet/blob/main/pkg/pipeline/crd.go) in CRD generator we generate a single type and in [group folder,](https://github.com/crossplane-contrib/terrajet/blob/main/pkg/pipeline/version.go) we don't have to know what types have been generated. If we wanted to have a `register.go` that has information of every type then we'd have to have a common bookkeeping, which would make it a bit harder to generate stuff in parallel, too. That's the case with ACK as well.

It's not too hard to build but it just seems not very worth it. Also when there are more than 10 CRDs in a package, `register.go` becomes harder to read.

Reply:
Actually I just noticed the name `types.go`. We store each CRD in different file in both terrajet and ACK like `<kind>_types.go`. What do you think about renaming it to `mytype_types.go`?

Reply:
@muvaf I like that a lot!